_posts/en/newsletters/2020-12-16-newsletter.md

```diff
@@ -0,0 +1,331 @@
+    (e.g. the ID of your peer), reconnect to the peer after you lose
+    data, and request the peer immediately close the channel.  This does
+    seem to signal that you've lost data and so your peer can close the
+    channel in an old state, but that isn't guaranteed that will
+    work---you might still have old states that will allow you to
+    penalize them.
+
+    This week, Lloyd Fournier started two threads on the Lightning-Dev
+    mailing list about possible improvements to the above mechanisms:
+
+    - **Fast recovery without backups:** the static per-channel backups
+      that allow fast recovery of funds require you create a new backup
+      each time you open a new channel.  If you fail to make a backup,
+      your only option is to wait until your channel counterparty
+      decides to close the channel on their own.  Fournier instead
+      [proposed][really static backups] a deterministic key derivation method that would allow a
+      node to search through the list of public LN nodes, combine
+      information about its private keys derived from its HD wallet with
+      information about each node's main public key, and determine
+      whether or not it had a channel with that node.  This would only
+      work for channels you open with public nodes, but it's expected
+      that's what most regular people will be using anyway.
+
+    - **Covert request for mutual close:** the existing mechanism for
+      closing a channel requires that your counterparty broadcast their
+      unilateral commitment transaction.  It would be better to use a
+      mutual close transaction---this uses less space onchain, requires
+      paying less fees, is not identifiable onchain as having belonged
+      to an LN channel, and allows both parties to spend their funds
+      immediately.  However, mutual close transactions don't contain any
+      penalty mechanism, so if you request a channel be closed and your
+      counterparty gives you an inaccurate mutual close transaction,
+      there's no way for you to penalize them.  In the normal protocol
+      this isn't an issue---you'd just broadcast the latest state, but
+      if you've lost your state, then you have no remedy.
+
+        Fournier proposed a solution using a cryptographic primitive
+        called [oblivious transfer][].  The technique allows you to
+        decide when you reconnect to your counterparty whether or not
+        you want to mutually close the channel.  If you do, they give
+        you a copy of the unsigned mutual close transaction plus an
+        encrypted copy of their signature for it.  You decrypt the
+        signature, provide your own signature, and broadcast the
+        transaction.  If you don't want to close the channel, you
+        perform all the same steps from their perspective until you
+        receive the encrypted signature.  You then prove to them that
+        generated the encryption key in a way that prevents you (or
```

Comment:
```suggestion
        they(?) generated the encryption key in a way that prevents you (or
```

Reply:
I'm having trouble following this sentence. I thought they gave me an encrypted copy of their signature, but now we prove to that we generated something. It seems that there is either a step missing or "you" and "they" switched sides. Otherwise, how would we ever be able to prove the absence of our ability to decrypt something?

Reply:
I agree that this is quite difficult to follow.

Reply:
Sorry, it's hard to compress two pages of protocol description to a single paragraph.  :-)  I pushed a revision that just focuses on the results.  But, to answer your question in case you're curious, the trick is that you choose and send them the encryption key.  If you want a copy of the mutual close transaction, you chose a regular working encryption key.  If you don't want the transaction, you chose a key for which you can prove you couldn't know the corresponding decryption key.